src/main/java/TaskManager.java

```diff
@@ -0,0 +1,113 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+
+public class TaskManager {
+
+    private static TaskManager instance = null;
+    private static HashMap<Integer,Task> storage;
+    
+    private TaskManager() {
+        storage = new HashMap<Integer,Task>();
+    }
+
+    /**
+     * New TaskManager instance. Handles all tasks.
+     */
+    public static TaskManager getInstance() {
+        if (instance == null) {
+            instance = new TaskManager();
+        }
+        return instance;
+    }
+
+    /**
+     * Saves current task.
+     */
+    public List<String> addTask(String taskType, String message) {
+        List<String> messages = new ArrayList<String>();
+        Task task;
+        int taskIndex = storage.size() + 1;
+        switch (taskType) {
+            case Constants.TODO:
+                task = new Todo(taskIndex, message, false);
+                break;
+            case Constants.DEADLINE:
+                String[] deadlineTask = message.split(" /by ");
+                String details = deadlineTask[0].trim();
+                if (deadlineTask.length <= 1 || details.isEmpty()) {
+                    messages.add(Constants.deadlineErrorMessage);
+                    return messages;
+                }
+                String time = deadlineTask[1].trim();
+                task = new Deadline(taskIndex, details, false, time);
+                break;
+            case Constants.EVENT:
+                String[] eventTask = message.split(" /at ");
+                details = eventTask[0].trim();
+                if (eventTask.length <= 1 || details.isEmpty()) {
+                    messages.add(Constants.eventErrorMessage);
+                    return messages;
+                }
+                time = eventTask[1].trim();
+                task = new Event(taskIndex, details, false, time);
+                break;
+            default:
+                messages.add(Constants.todoErrorMessage);
+                return messages;
+        }
```

Comment:
I think there is an extra indentation for all 'case' within the switch statement.